src/components/course/data/hooks.jsx

```diff
@@ -447,50 +447,12 @@ export const useTrackSearchConversionClickHandler = ({ href, eventName }) => {
   return handleClick;
 };
 
-/**
- * Returns a function to be used as a click handler that emits an analytics event for a
- * search conversion via ``sendEnterpriseTrackEvent``. when used on a local hyperlink
- *
- * @param {object} args
- * @param {string} args.href If click handler is used on a hyperlink, this is the destination url.
- * @param {string} args.eventName Name of the event
- *
- * @returns Click handler function for clicks on buttons
- *  */
-export const useTrackSearchConversionClickHandlerLocal = ({ eventName }) => {
-  const {
-    state: {
-      activeCourseRun: { key: courseKey },
-      algoliaSearchParams,
-    },
-  } = useContext(CourseContext);
-  const { enterpriseConfig } = useContext(AppContext);
-  const handleClick = useCallback(
-    () => {
-      const { queryId, objectId } = algoliaSearchParams;
-      sendEnterpriseTrackEvent(
-        enterpriseConfig.uuid,
-        eventName,
-        {
-          products: [{ objectID: objectId }],
-          index: getConfig().ALGOLIA_INDEX_NAME,
-          queryID: queryId,
-          courseKey,
-        },
-      );
-    },
-    [algoliaSearchParams, enterpriseConfig, eventName, courseKey],
-  );
-
-  return handleClick;
-};
-
 /**
  * Returns a function to be used as a click handler that emits an optimizely enrollment click event.
  *
  * @returns Click handler function for clicks on enrollment buttons.
  */
-export const useOptimizelyEnrollmentClickHandler = ({ href, courseRunKey, courseEnrollmentsByStatus }) => {
+export const useOptimizelyEnrollmentClickHandler = ({ href = undefined, courseRunKey, courseEnrollmentsByStatus }) => {
   const enrollmentCountIsZero = Object.values(courseEnrollmentsByStatus).flat().length === 0;
```

Comment:
Interesting. I wonder if implementation would be any simpler if instead of relying on `courseEnrollmentsByStatus`, we relied on the `userEnrollments` passed through the `CourseContext` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/a5754ba055311a73bcaa0d60c489bea7748af27a/src/components/course/CoursePage.jsx#L156)).

Given the use of `courseEnrollmentsByStatus` here doesn't really need the grouping "by status" like the dashboard route does, I believe we could rely on the simpler `userEnrollments` array here instead, i.e.:

```jsx
const hasNoExistingEnrollments = userEnrollments.length === 0;
```

I believe a benefit of this change would be that we'd no longer need to rely on the `CourseEnrollmentsContext` value where we call `useOptimizelyEnrollmentClickHandler`, instead using an already flat array of `userEnrollments` from `CourseContext` instead, which should also be accessible.